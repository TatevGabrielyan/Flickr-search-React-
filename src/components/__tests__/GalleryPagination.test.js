import React from 'react';
import ReactDOM from 'react-dom';
import TestUtils from 'react-dom/test-utils';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';

import GalleryPagination from '../GalleryPagination';
import state from '../../config/mockState';
import store from '../../config/mockStore';
import { SEARCH_FETCH_REQUEST } from '../../actions';

describe('GalleryPagination component', () => {
  const { search } = state;
  const { text, pageSelected } = search;

  const component = (
    <Provider store={store}>
      <GalleryPagination />
    </Provider>
  );

  beforeEach(() => {
    store.clearActions();
  });

  it('Renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(component, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('Snapshot matchs', () => {
    const tree = renderer.create(component).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Click on previous button', () => {
    const tree = TestUtils.renderIntoDocument(component);
    const buttons = TestUtils.scryRenderedDOMComponentsWithTag(tree, 'button');
    const previousButton = buttons[0];

    //Call the function
    TestUtils.Simulate.click(previousButton);

    const expectedActions = [
      {
        type: SEARCH_FETCH_REQUEST,
        payload: {
          meta: {
            text,
            page: pageSelected - 1,
          },
        },
      },
    ];
    expect(store.getActions()).toEqual(expectedActions);
  });

  it('Click on next button', () => {
    const tree = TestUtils.renderIntoDocument(component);
    const buttons = TestUtils.scryRenderedDOMComponentsWithTag(tree, 'button');
    const nextButton = buttons[1];

    //Call the function
    TestUtils.Simulate.click(nextButton);

    const expectedActions = [
      {
        type: SEARCH_FETCH_REQUEST,
        payload: {
          meta: {
            text,
            page: pageSelected + 1,
          },
        },
      },
    ];
    expect(store.getActions()).toEqual(expectedActions);
  });
});
