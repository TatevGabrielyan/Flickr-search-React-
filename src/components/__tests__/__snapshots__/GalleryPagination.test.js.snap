// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GalleryPagination component Snapshot matchs 1`] = `
<div
  className="GalleryPagination"
>
  <button
    className="GalleryPaginationButton"
    disabled={false}
    onClick={[Function]}
    type="button"
  >
    Previous
  </button>
  <span
    className="GalleryPaginationText"
  >
    Page 
    5
     of 
    3710
     
  </span>
  <button
    className="GalleryPaginationButton"
    onClick={[Function]}
    type="button"
  >
    Next
  </button>
</div>
`;
